### PR TITLE
Implement ReadModifyWriteRule::IncrementAmount()

### LINKED ISSUE #302: Implement bigtable::ReadModifyWriteRule::IncrementAmount()
This depends on #300, and must include the unit test, of course.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #302.

### CODE REVIEW

**Comment by coryan** [comment]
**tl;dr;** this is how we defined the wrappers to create all the protos in the client library, check go/cbt-cpp:surface-dd, it would be bad to make only this proto different.

> Is this interface (two statics and a getter returning the raw, underlying object) getting a bit peculiar?

We have been using this (admittedly peculiar) style for all the proto builder interfaces.  The are typically used like so:

```C++
using R = bigtable::ReadModifyWriteRule;
table->ReadModifyWrite("my-row", R::AppendValue("f", "c", "new-value"), R::IncrementAmount("fam", "counter", 42));
```

You can look at the `bigtable::Filter` class for more complex examples.

> Would a private constructor be better than twice calling the two getters?

Well, protos do not have constructors that initialize each member (don't get me started), so the private constructor would have to call each setter in turn.

> Would the statics be better as methods at this stage, or constructors)?

Constructors could be used in this case, because the overloads are different, but we have used this static functions in other wrappers because protos are used as discriminated unions, but with the same types.  For example, these two return filters with different values:

```C++
using F = bigtable::Filter;
auto pass_all = F::PassAllFilter();
auto block_all = F::BlockAllFilter();
```

I mean, we could use a constructor with an extra type that is used as the discriminant or something, like so:

```C++
// Assuming struct PassAllFilter; and struct BlockAllFilter; where defined earlier...
bigtable::Filter pass_all(PassAllFilter{});
bigtable::Filter pass_all(BlockAllFilter{});
```

I think the former is (marginally) more readable, and composes better, e.g., you can say things like this today:

```C++
table->ReadRows(...., F::Chain(F::Latest(1), F::ColumnRange("fam", "col0", "col9"), F::ValueRegex("foo.*"));
```

Anyway, if you feel strongly about it we would need to change all the other APIs to avoid confusion.

**Comment by richdutton** [comment]
Thanks a lot for the thoughtful, informative response. In that case LGTM
